integration-tests/tests/src/tests/schema.ts

```diff
@@ -44,6 +44,32 @@ describe("Realm schema", () => {
         return realm.create<Test>("Test", {});
       });
 
+      expect(test.dynamic).to.equal(42);
+    });
+    it("can take a function as a default property value for a primary key", () => {
+      interface Test {
+        dynamic?: number;
+      }
+
+      const realm = new Realm({
+        schema: [
+          {
+            name: "Test",
+            primaryKey: "dynamic",
+            properties: {
+              dynamic: {
+                type: "int",
+                default: () => 42,
+              },
+            },
+          },
+        ],
+      });
```

Comment:
Any reason this can't use the `openRealmBefore` hook?

Reply:
I've refactored the tests